File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int total = 0;
        int temp = numberToCheck;
        int n = (int) Math.log10(numberToCheck) + 1;

        while (temp > 0) {
            total += Math.pow(temp % 10, n);
            temp /= 10;
        }

        return total == numberToCheck;
    }

}

File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"

#include <math.h>

bool is_armstrong_number(int candidate) {
    int sum = 0;
    int temp = candidate;
    int digit;
    int num_length = floor(log10(candidate)) + 1;
    
    while (temp > 0) {
        digit = temp % 10;
        sum += pow(digit, num_length);
        temp /= 10;
    }

    return sum == candidate;
}